src/main/java/org/opentripplanner/routing/core/RoutingContext.java

```diff
@@ -222,4 +201,32 @@ public void checkIfVerticesFound() {
     public void destroy() {
         this.tempEdges.forEach(DisposableEdgeCollection::disposeEdges);
     }
+
+
+    /* PRIVATE METHODS */
+
+    private boolean isDisconnected(Set<Vertex> vertices, boolean isFrom) {
+        // Not connected if linking was not attempted, and vertices were not specified in the request.
+        if (vertices == null) { return true; }
+
+        Predicate<Vertex> isNotTransit = Predicate.not(TransitStopVertex.class::isInstance);
+        Predicate<Vertex> hasNoIncoming = v -> v.getIncoming().isEmpty();
+        Predicate<Vertex> hasNoOutgoing = v -> v.getOutgoing().isEmpty();
+
+        // Not connected if linking did not create incoming/outgoing edges depending on the
+        // direction and the end.
+        Predicate<Vertex> isNotConnected = isFrom ^ opt.arriveBy
```

Comment:
This took me a moment to understand because 
1. `^` is the bitwise non-short-circuit XOR operator, which I'm not used to seeing used with booleans
2.  I didn't immediately catch that operator precedence would group the two booleans with this operator, so at first I was wondering what would happen when performing a bitwise XOR between a boolean and a predicate object reference.

It seems preferable to use parentheses and spell out the cases in terms of more commonly used logical operators && and ||. However that gets long and repetitive. I tried to refactor several ways and didn't come up with anything better, so I think we should at least add some parens and clarify a bit in the comment line.

```suggestion
        // direction and the end. Use bitwise XOR operator because there is no separate logical XOR.
        Predicate<Vertex> isNotConnected = (isFrom ^ opt.arriveBy)
```

Reply:
This was simplified in https://github.com/opentripplanner/OpenTripPlanner/pull/3628/commits/d5f34519a5eea4d85825fcf6ea8a9669802c7fe5